kilkari/src/main/java/org/motechproject/nms/kilkari/service/ChildCsvThreadProcessor.java

```diff
@@ -54,21 +51,64 @@ public ThreadProcessorObject call() throws Exception { //NOPMD SignatureDeclareT
         String id;
         String contactNumber;
         String childInstance;
+        String mctsIdForRchChild=null;
+        String motherId;
+        String mctsMotherIdForChild=null;
         if (mctsImport) {
             id = KilkariConstants.BENEFICIARY_ID;
             contactNumber = KilkariConstants.MSISDN;
             childInstance = KilkariConstants.MCTS_CHILD;
+            motherId=KilkariConstants.MOTHER_ID;
         } else {
             id = KilkariConstants.RCH_ID;
             contactNumber = KilkariConstants.MOBILE_NO;
             childInstance = KilkariConstants.RCH_CHILD;
+            mctsIdForRchChild=KilkariConstants.MCTS_ID;
+            motherId=KilkariConstants.RCH_MOTHER_ID;
+            mctsMotherIdForChild=KilkariConstants.MCTS_MOTHER_ID;
         }
         int count = 0;
         Timer timer = new Timer("kid", "kids");
         for(Map<String, Object> record : recordList) {
             count++;
             LOGGER.debug("Started child import for msisdn {} beneficiary_id {}", record.get(contactNumber), record.get(id));
 
+            //filter for  rch child's Registration_No and mcts child's ID_no
+            String newRchId=(String)record.get(id);
+            newRchId=newRchId.replaceAll("[\\n\\t\\r ]","");
+            record.replace(id,newRchId);
+            //filter for rch child's MCTS_Mother_ID_No
+
+            if(!mctsImport) {
+                //for rch child's mother registration no
+                try {
+                    String childLinkedMother=(String)record.get(motherId);
+                    childLinkedMother=childLinkedMother.replaceAll("[\\n\\t\\r ]","");
+                    record.replace(motherId,childLinkedMother);
+                }
+                catch (Exception e){
+                    //LOGGER.debug("no mother for child");
+                }
+                //for rch child's MCTS_ID_No
+                try {
+                    String newMctsId=(String)record.get(mctsIdForRchChild);
+                    newMctsId=newMctsId.replaceAll("[\\n\\t\\r ]","");
+                    record.replace(mctsIdForRchChild,newMctsId);
+                }
+                catch (Exception e){
+                    //LOGGER.debug("no mcts id for child");
+                }
+                // for rch child's MCTS_Mother_Id_N0
+                try{
+                    String newMctsMotherId=(String)record.get(mctsMotherIdForChild);
+                    newMctsMotherId=newMctsMotherId.replaceAll("[\\n\\t\\r ]","");
+                    record.replace(mctsMotherIdForChild,newMctsMotherId);
+                }
+                catch (Exception e){
+                    //LOGGER.debug("no mcts mother id for child");
+                }
+            }
+
```

Comment:
there are four replace logic, which can be abstracted out in a single method and the same method can be reused.
Also, just make sure there is no scope of null pointer in (String)record.get(xyz).